### ISSUE TITLE
Even though all projects are set with workspace resolution disabled, editing parent pom causes numerous projects to rebuild

### ISSUE THREAD

**Issue by davidmichaelkarr** [eclipse-m2e/m2e-core#1368]
Using Eclipse 2022-03.

I always have "Workspace Resolution" DISABLED in every project in Eclipse, because I want them to get their dependencies from the local repo, not from a workspace project.  I even wrote a shell script that modifies the .settings file in all projects, just to make sure I didn't miss one.

Even with that, when I make changes to the parent pom, even before I save them, it results in many of my workspace projects suddenly starting to build, which takes a long time to finish. I sometimes have the presence of mind to turn off "Build Automatically" before I do that, which I think helps, but I'm not certain.

**Comment by davidmichaelkarr** [comment]
I have confirmed that even with "Workspace Resolution" disabled in all projects, and "Build Automatically" UNchecked, when I save the pom.xml for a parent pom, it causes many projects that reference that parent pom to automatically start builds.

**Comment by HannesWell** [comment]
> Using Eclipse 2022-03.

This is a relatively old version of Eclipse, can you try it with the latest one 2023-03?
But AFAIKT nothing changed in that area recently, can you tell if this worked before?
@laeubi are you aware of any change that could result this behavior?

A minimal test case for https://github.com/eclipse-m2e/m2e-core/tree/master/org.eclipse.m2e.core.tests that demonstrates the issue would help to identify the cause. Maybe you can then also debug m2e to find it?

**Comment by davidmichaelkarr** [comment]
Sorry, typo. I meant 2023-03.

**Comment by davidmichaelkarr** [comment]
And to the other pojnt, I had noticed this was happening a few weeks ago, but I wasn't paying enough attention to really realize it. I don't often work on parent poms.

**Comment by laeubi** [comment]
@davidmichaelkarr Please open the "Maven Workspace Build" view, then enable recording and you should see what is triggering an (automatic) build.

**Comment by davidmichaelkarr** [comment]
My workspace had finally settled down, so I opened that view, which I've never noticed before.  I turned on "Build Automatically", and I immediately saw output in the Maven Console, showing "Downloading" and "Downloaded" reports, but there is still nothing in the MWB view. In the "Progress" view, I see an entry for "Invoking Maven Project Builder" for one project in my workspace. Still nothing in MWB.

I'm still watching the Maven Console view printing log statements. Besides the "Downloading" and "Downloaded" entries, I see lines like this:

    Using org.eclipse.m2e.jdt.JarLifecycleMapping lifecycle mapping for MavenProject: com.att.idp:idp-config:2.8.0 @ C:\Users\dk068x\git\idp-config\pom.xml.

That represents one of the many projects in my workspace, and it is a library referenced by other poms in the workspace. I'm seeing similar messages associated with other projects in the workspace.

When I first turned on "Build Automatically" in this session, I think it said "Building 80%" in the status bar. After about ten minutes, it's now at 93%.

It then jumped down to "Building 10%", still pumping logs in the Maven Console view. Not at "Building 22%".

I'm seeing some " [ERROR] Unexpected URLConnection defaultUseCaches enabled" lines in that log.

Although I don't see any duplicate entries (in entries with today's date) of the "Using ..." log entry, I do see many duplicates of "Downloading" and "Downloaded" pairs for common plugins, like "maven-javadoc-plugin" and "maven-source-plugin". I see those two repeated extremely frequently.  I also see those pairs for the artifacts corresponding to some of my workspace projects. Now at "Building 80%" again. At this point, it's been perhaps 20 minutes since I turned on "Build Automatically", and it's still going. There still is nothing in the MWB view.

And then it finally finished. It took about 25 minutes, and that was without making any changes at all in my parent poms.  Later this morning I will try another experiment of making a change to the parent pom.

**Comment by davidmichaelkarr** [comment]
This morning, I saved a change to a BOM that is imported by our parent poms. It instantly started a series of builds of my workspace projects, all of which have Workspace Resolution disabled. I still have never seen anything appear in the MWB view.

In the Maven Console view, I see lines like this:

    5/6/23, 9:02:12 AM PDT: [INFO] Using NULL lifecycle mapping for MavenProject: com.att.idp:sdk-java-parent:2.8.0 @ C:\Users\dk068x\git\sdk-java-parent\pom.xml.
    5/6/23, 9:02:12 AM PDT: [INFO] Using org.eclipse.m2e.jdt.JarLifecycleMapping lifecycle mapping for MavenProject: com.att.idp:PlatformPilotMs:1.1.0-SNAPSHOT @ C:\Users\dk068x\git\platformpilotms\pom.xml.

**Comment by laeubi** [comment]
Yes the delta would be the interesting thing because that means the project was build because m2e detected some change in the project, if you only see downloading messages, it might be that the download source/javadoc is kicking in.

Also it _might_ be that m2e still updates related projects but just resolve not the dependencies, that are different processes, so a small example would help to investigate this, in the meanwhile you probably want to disable the "automatic update on project change" in the m2e preferences.

**Comment by davidmichaelkarr** [comment]
I assume you mean "Automatically update Maven projects configuration"?  What exactly does that do?

**Comment by laeubi** [comment]
It does not update project configuration if a pom changes (you can do that manually with the menu Maven > Update Project configaration).
